optimize/newton.go

```diff
@@ -152,7 +152,7 @@ func (n *Newton) NextDirection(loc *Location, dir []float64) (stepSize float64)
 		pd := n.chol.Factorize(n.hess)
 		if pd {
 			// Store the solution in d's backing array, dir.
-			n.chol.SolveVecTo(d, grad)
+			_ = n.chol.SolveVecTo(d, grad)
```

Comment:
Are we comfortable with this? Does it deserved a comment?

Reply:
perhaps introduce a `must` function?
```go
func must(err error) {
    if err != nil {
        panic(err)
    }
}
```
and use that? (should be inlined by the compiler.)

Reply:
The issue is not that, but whether this is guaranteed to not return an error. If it can, then we need to consider what to do. Whether panicking is OK is sort of the question there; if it is, then I'd just do a conditional panic.

Reply:
I'd say this should check the error:
```
err := n.chol.SolveVecTo(d, grad)
if err == nil {
    d.ScaleVec(-1, d)
    return 1
}
```
`Cholesky.SolveVecTo` returns an error only if the modified Hessian is badly conditioned. It will compute something but  it may not be a descent direction, leading to an `ErrNonDescentDirection`. This modification increases the chance of overcoming this error at the expense of an extra Cholesky factorization.

Reply:
Done